### PR TITLE
docs(spanner): update description of the timeout_secs parameter

### PR DESCRIPTION
**Pull request by Emar-Kar**
Closes: #9364

### CODE REVIEW

File: spanner/google/cloud/spanner_v1/database.py [hunk 1]
```diff
@@ -416,9 +416,10 @@ def run_in_transaction(self, func, *args, **kw):
         :param args: additional positional arguments to be passed to ``func``.
 
         :type kw: dict
-        :param kw: optional keyword arguments to be passed to ``func``.
+        :param kw: (Optional) keyword arguments to be passed to ``func``.
                    If passed, "timeout_secs" will be removed and used to
-                   override the default timeout.
+                   override the default timeout which defines maximum timestamp
```
Comment (larkee) [inline spanner/google/cloud/spanner_v1/database.py]: Perhaps reword it to `override the retry timeout` as `default timeout` is still vague and not particular helpful.

File: spanner/google/cloud/spanner_v1/session.py [hunk 2]
```diff
@@ -273,9 +273,10 @@ def run_in_transaction(self, func, *args, **kw):
         :param args: additional positional arguments to be passed to ``func``.
 
         :type kw: dict
-        :param kw: optional keyword arguments to be passed to ``func``.
+        :param kw: (Optional) keyword arguments to be passed to ``func``.
                    If passed, "timeout_secs" will be removed and used to
-                   override the default timeout.
+                   override the default timeout which defines maximum timestamp
```
Comment (larkee) [inline spanner/google/cloud/spanner_v1/session.py]: See above.

File: spanner/google/cloud/spanner_v1/database.py (hunk 1 above)
Comment (Emar-Kar) [reply spanner/google/cloud/spanner_v1/database.py]: I got your point and I think it better be called `the default retry timeout`.

Comment (larkee) [reply spanner/google/cloud/spanner_v1/database.py]: SGTM
